cypress_shared/pages/apply/confirmDetails.ts

```diff
@@ -11,6 +12,14 @@ export default class ConfirmDetailsPage extends Page {
     cy.get('dl').within(() => {
       this.assertDefinition('Name', this.person.name)
       this.assertDefinition('CRN', this.person.crn)
+      this.assertDefinition('Date of Birth', DateFormats.isoDateToUIDate(this.person.dateOfBirth))
+      this.assertDefinition('NOMS Number', this.person.nomsNumber)
+      this.assertDefinition('Nationality', this.person.nationality)
+      this.assertDefinition('Religion or belief', this.person.religionOrBelief)
+      this.assertDefinition('Sex', this.person.sex)
```

Comment:
quite likely I'm not understanding properly how these tests are wired up, but are we missing the demonstration that the `removeBlankSummaryListItems` function is being used to suppress empty lines? And could we do that with a negative assertion here? 

I'm not clear where/how the `person` picture is injected into the test but could we:

1. declare that `person.genderIdentity` is `null`
2. assert `this.assertNoDefinition('Gender identity')`

Reply:
This is a tricky one, we're trying not to dig down too much into individual sub-scenarios and focus on the happy paths, as Cypress tests tend to add a fair amount of overhead in terms of test run size. 

What I have done in the past when there's logic in views is test the unit test the views themselves  (see https://github.com/UKGovernmentBEIS/regulated-professions-register/blob/main/views/professions/show.njk.spec.ts), but again, I'm not sure how valuable this is.

Cypress has just introduced [Component Testing](https://docs.cypress.io/guides/component-testing/writing-your-first-component-test), which seems like a good candidate, but this seems more focussed on client-side apps.